### DISCUSSION TITLE
Getting Depth Images

### DISCUSSION THREAD

**Discussion by jueunkwon** [microsoft/AirSim#3655] [Support Q&A]
Hello! Currently when I record while I am flying a drone in the forest environment, scene images get saved in the AirSim folder. 
Is there a way to get the depth images saved in the AirSim folder as well? Thank you!!

**Comment by rajat2004** [comment]
You can modify the [Recording settings](https://microsoft.github.io/AirSim/settings/#recording), see the example given and the default ones as well. In particular you'll need to add a new element in Cameras with the appropriate [ImageType](https://microsoft.github.io/AirSim/image_apis/#available-imagetype-values)

**Reply by jueunkwon** [reply]
Thank you for the quick response. I have updated the settings to get both ImageType 0 and 2. However only the airsim_rec gets saved, and either of the image types are being saved. It is also likely that there is an error in my settings so I attached my settings below. 
I want to be able to get depth and Scene views simultaneously like the ones from https://github.com/microsoft/AirSim/issues/244#issuecomment-316549284 but with record. Thank you. 
```
{
  "SeeDocsAt": "https://github.com/Microsoft/AirSim/blob/master/docs/settings.md",
  "SettingsVersion": 1.2,

  "SimMode": "Multirotor",

  "Recording": {
    "RecordOnMove": false,
    "RecordInterval": 0.05,
    "Folder": "",
    "Enabled": false,
    "Cameras": [
      {
        "CameraName": "0",
        "ImageType": 0,
        "PixelsAsFloat": false,
        "VehicleName": "",
        "Compress": true
      },
      {
        "CameraName": "0",
        "ImageType": 2,
        "PixelsAsFloat": false,
        "VehicleName": "",
        "Compress": true
      }
    ]
  }
}
```

**Reply by rajat2004** [reply]
That's strange, just tested using the settings you gave and both Scene and Depth images are getting recorded. Do you mean only one is them is getting saved, or none? Which version of AirSim are you using? If not already, try the latest release

**Reply by jueunkwon** [reply]
None of them are getting recorded, and I am using the latest release. Thank you!

**Reply by jueunkwon** [reply]
Are there any other steps I am supposed to take after editing the settings.json code? I thought it was suppose to work automatically when I click run inside unreal engine since it calls settings.json from there. Are there are any other edits that I have to do such as editing the API?

**Reply by rajat2004** [reply]
Sorry, had some work so forgot. The platform is Windows or Linux? Which env is this problem occurring with, and please test the Blocks env as well. There were some problems with a few envs which have been fixed, and it's unlikely that such an issue has gone unnoticed till now. Make sure that there are no other `settings.json` as well located somewhere. Also please upload the `airsim_rec.txt`, to see if contains the image names which should have been saved.

**Reply by jueunkwon** [reply]
Hello, thank you for all the help. I really appreciate it. I am using Windows. Everything works fine when I use the default setting by leaving just the "SettingsVersion": 1.2 in the settings.json file, but if I edit anything in the file, although I do get the txt file, I do not get any images.  I attached my txt file for the current json code above here: [airsim_rec.txt](https://github.com/microsoft/AirSim/files/6489851/airsim_rec.txt). Thank you so much!
